### PR TITLE
Add join games

### PR DESCRIPTION
**Pull request by zenkbaries**
Added column to games/index.html.erb for 'join' link.

Added code to games#update to add current user to black_player_id.

Added navbar:/ signin/signout/signup, 'join game' link 

Notes:
there is no column for black_player_in in games model. Need to add 'black_player_id' column to game modal?

### CODE REVIEW

File: app/views/games/index.html.erb [hunk 1]
```diff
@@ -7,17 +7,21 @@
...
-      <tr>
-        <td><%= game.name %></td>
-        <td><%= link_to 'Show', game %></td>
-        <td><%= link_to 'Edit', edit_game_path(game) %></td>
-        <td><%= link_to 'Destroy', game, method: :delete, data: { confirm: 'Are you sure?' } %></td>
-      </tr>
+      if game.available? do
+        <tr>
+          <td><%= game.name %></td>
+          <td><%= link_to 'Join', games, method: :update %></td>
```
Comment (mscoutermarsh) [inline app/views/games/index.html.erb]: Looking at this I have an idea :bulb: :)

For this button, there is a way to pass query parameters along as well.

See if you can update it to send `game: { black_player_id: current_user.id }` to the controller.

Then you wouldn't need to change the controller at all (could keep the existing code that was there before)

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -15,6 +15,57 @@
...
+          <ul class="dropdown-menu">
+            <li><a href="#">Action</a></li>
+            <li><a href="#">Another action</a></li>
+            <li><a href="#">Something else here</a></li>
+            <li role="separator" class="divider"></li>
+            <li><a href="#">Separated link</a></li>
+            <li role="separator" class="divider"></li>
+            <li><a href="#">One more separated link</a></li>
+          </ul>
+        </li> -->
```
Comment (mscoutermarsh) [inline app/views/layouts/application.html.erb]: if we don't need this code, can be removed

**Comment by mscoutermarsh** [comment]
If we missed `black_player_id` when creating the original migrations.

Could you add a migration to add it? Also double check on `white_player_id`?

File: db/schema.rb [hunk 3]
```diff
@@ -17,11 +17,15 @@
   enable_extension "plpgsql"
 
   create_table "games", force: true do |t|
-    t.string   "name"
+    t.integer  "game_id"
+    t.integer  "white_player"
+    t.integer  "black_player"
```
Comment (mscoutermarsh) [inline db/schema.rb]: Also, to add a column. You should be creating a migration and using `add_column`. So there should be a migration file as well.

Schema.rb is autogenerated

**Comment by zenkbaries** [comment]
i noticed the conflict is "schema.rb" the only conflict. I know that schema.rb is autogenerated so how do I handle it?  i just choose my part of it and let the schema.rb rebuilt itself.  
let me know if this is not the right move or not.  want to get that branch merged and move on.

**Pull request merged**